pkg/secret/secret.go

```diff
@@ -0,0 +1,165 @@
+package secret
+
+import (
+	"fmt"
+	"github.com/appuio/seiso/pkg/kubernetes"
+	"github.com/appuio/seiso/pkg/openshift"
+	"github.com/appuio/seiso/pkg/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sort"
+	"time"
+)
+
+type SecretsService struct {
+	client core.SecretInterface
+	helper kubernetes.Kubernetes
+}
+
+type Service interface {
+	ListNamesAndLabels() (resourceNames, labels []string, err error)
+	List(listOptions metav1.ListOptions) (resources []v1.Secret, err error)
+	GetUnused(namespace string, secrets []v1.Secret) (unusedSecrets []v1.Secret, funcErr error)
+	Delete(secrets []v1.Secret)
+	FilterByTime(secrets []v1.Secret, olderThan time.Time) (filteredSecrets []v1.Secret)
+	FilterByMaxCount(secrets []v1.Secret, keep int) (filteredSecrets []v1.Secret)
+	Print(secrets []v1.Secret, batch bool)
+}
+
+//Constructor
+func NewSecretsService(client core.SecretInterface, helper kubernetes.Kubernetes) Service {
```

Comment:
Better comment: `// NewSecretsService creates a new Service instance`. Any static analysis tools like CodeClimate etc. would complain about this (they'd expect the comment to be in the form of `<function-name> <verb> <action-description>`) just like GoDoc.